runtime/src/stakes.rs

```diff
@@ -382,6 +369,33 @@ impl Stakes<StakeAccount> {
         }
     }
 
+    fn override_stake_accounts(&mut self, thread_pool: &ThreadPool, stake_rewards: &[StakeReward]) {
+        let stake_delegations: Vec<_> = thread_pool.install(|| {
+            stake_rewards
+                .into_par_iter()
+                .filter_map(|stake_reward| {
+                    let stake_account = StakeAccount::try_from(stake_reward.stake_account.clone());
+                    Some((stake_reward.stake_pubkey, stake_account.ok()?))
+                })
+                .collect()
+        });
+        self.stake_delegations = std::mem::take(&mut self.stake_delegations)
+            .into_iter()
+            .chain(stake_delegations)
+            .collect::<HashMap<Pubkey, StakeAccount>>()
+            .into_iter()
+            .filter(|(_, account)| account.lamports() != 0u64)
+            .collect();
+        let stake_delegations: Vec<_> = self.stake_delegations.values().collect();
+        self.vote_accounts = refresh_vote_accounts(
+            thread_pool,
+            self.epoch,
+            &self.vote_accounts,
+            &stake_delegations,
+            &self.stake_history,
+        );
```

Comment:
Did you mean to add this here? I'm guessing this is related to making `Bank::store_vote_accounts()` work in parallel for later work, but unless I'm missing something, this looks unnecessary since `bank.store_vote_accounts()` happens right after `bank.store_stake_accounts()`.

If you want to make `store_vote_accounts()` work in parallel, it would be better to do it in a separate PR, probably with some new `update_vote_accounts()` function on the stakes cache.

Reply:
`refresh_vote_accounts` only updates delegated stake to each vote-account, which is necessary here because we are updating stake accounts so the delegated stake to some vote-accounts might change. Similar to how `upsert_stake_delegation` updates the delegated stake on the associated vote-account:
https://github.com/solana-labs/solana/blob/16ec9a68c/runtime/src/stakes.rs#L370-L382

`store_vote_accounts` does not re-compute the delegated-stake. You see in `upsert_vote_account` it takes the delegated stake from the old vote-account it is updating:
https://github.com/solana-labs/solana/blob/16ec9a68c/runtime/src/stakes.rs#L358-L361

The delegated stake is updated each time a stake-account is upserted.

Reply:
Ah right, of course, I knew I missed something, my bad. Thanks for explaining!